datadog_checks_base/datadog_checks/base/stubs/datadog_agent.py

```diff
@@ -3,41 +3,38 @@
 # Licensed under a 3-clause BSD style license (see LICENSE)
 
 
-def get_hostname():
-    return 'stubbed.hostname'
+class DatadogAgentStub(object):
+    def __init__(self):
+        self._metadata = {}
 
+    def reset(self):
+        self._metadata = {}
 
-def log(*args, **kwargs):
-    pass
+    def assert_metadata(self, check_id, data):
+        for name, value in data.items():
+            assert self._metadata[(check_id, name)] == value
```

Comment:
The assert output may be more useful is we compare the whole dict at once. Some like:
```
assert data == {k,v for k,v in self._metadata.items() if k in data}
```

or maybe https://stackoverflow.com/a/41579450